cluster/rpc_test.go

```diff
@@ -251,6 +424,57 @@ func TestConnectionBalancer_roundRobinSelect(t *testing.T) {
 	})
 }
 
+func TestConnectionBalancer_mesh_network(t *testing.T) {
+	node, close := testRPCServer(t)
+	defer close()
+	node2, close := testRPCServer(t)
+	defer close()
+	mock := newMockRegistry([]Node{node, node2})
+
+	cfg := &ConnConfig{
+		MaxConnections:     0,
+		InitialNodeTimeout: testConnConfig.InitialNodeTimeout,
+		DebounceTime:       testConnConfig.DebounceTime,
+	}
+	c, err := newConnectionBalancer("", "foo", &mock, cfg)
+	require.NoError(t, err)
+	defer c.Close()
+
+	require.Equal(t, []Node{node, node2}, c.getSelectedNodes())
+
+	node3, close := testRPCServer(t)
+	defer close()
+	node4, close := testRPCServer(t)
+	defer close()
+
+	go func() {
+		mock.nodeChan <- []Node{node, node2, node3}
```

Comment:
Is there a test if MaxConnections isn't 0? To see what would happen with this during `getSelectedNodes`

Reply:
Yep, the rest of the tests use the test config which uses 3 max connections